das/singleton_das.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package das
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"os"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/offchainlabs/nitro/arbstate"
+	"github.com/offchainlabs/nitro/blsSignatures"
+	flag "github.com/spf13/pflag"
+)
+
+type SingletonDASConfig struct {
+	KeyDir             string `koanf:"key-dir"`
+	PrivKey            string `koanf:"priv-key"`
+	AllowGenerateKeys  bool   `koanf:"allow-generate-keys"`
+	StoreSignerAddress string `koanf:"store-signer-address"`
+}
+
+func SingletonDASConfigAddOptions(prefix string, f *flag.FlagSet) {
+	f.String(prefix+".key-dir", "", fmt.Sprintf("The directory to read the bls keypair ('%s' and '%s') from", DefaultPubKeyFilename, DefaultPrivKeyFilename))
+	f.String(prefix+".priv-key", "", "The base64 BLS private key to use for signing DAS certificates")
+	f.Bool(prefix+".allow-generate-keys", false, "Allow the local disk DAS to generate its own keys in key-dir if they don't already exist")
+	f.String(prefix+".store-signer-address", "", "Address required to sign stores, or empty if anyone can store")
+}
+
+type SingletonDAS struct {
```

Comment:
I'm not sure I understand the difference between this and the `LocalDiskDAS`. They seem quite similar

Reply:
Combined them by extending `LocalDiskDAS` and then deleting `SingletonDAS`.